qiskit/primitives/utils.py

```diff
@@ -61,7 +61,7 @@ def init_observable(observable: BaseOperator | str) -> SparsePauliOp:
     if isinstance(observable, SparsePauliOp):
         return observable
     elif isinstance(observable, BaseOperator) and not isinstance(observable, BasePauli):
-        return SparsePauliOp.from_operator(observable)
+        raise QiskitError(f"observable type not supported: {type(observable)}")
```

Comment:
This was directly introduced into 0.46: https://github.com/Qiskit/qiskit/pull/11535